LeadKit/LeadKit/Extensions/UIImage/UIImage+Gradients.swift

```diff
@@ -0,0 +1,123 @@
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let colors = gradientColors.map {(color: UIColor) -> AnyObject? in return color.CGColor as AnyObject? } as NSArray
+        let gradient = CGGradientCreateWithColors(colorSpace, colors, nil)
+        CGContextDrawLinearGradient(context,
+                                    gradient,
+                                    CGPoint(x: 0, y: 0),
+                                    CGPoint(x: 0, y: size.height),
+                                    CGGradientDrawingOptions(rawValue: 0))
+        guard let cgImage = UIGraphicsGetImageFromCurrentImageContext().CGImage else {
+            return nil
+        }
+        self.init(CGImage: cgImage)
+        UIGraphicsEndImageContext()
+    }
+
+    /**
+     applies gradient color overlay to an image.
+
+     - parameter gradientColors: An array of colors to use for the gradient.
+     - parameter blendMode: The blending type to use.
+
+     - returns: A new image
+     */
+    public func applyGradientColors(gradientColors: [UIColor],
+                             blendMode: CGBlendMode = .Normal) -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(size, false, scale)
+        let context = UIGraphicsGetCurrentContext()
+        CGContextTranslateCTM(context, 0, size.height)
+        CGContextScaleCTM(context, 1.0, -1.0)
+        CGContextSetBlendMode(context, blendMode)
+        let rect = CGRect(x: 0, y: 0, width: size.width, height: size.height)
+        CGContextDrawImage(context, rect, CGImage)
+        // Create gradient
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let colors = gradientColors.map {(color: UIColor) -> AnyObject? in return color.CGColor as AnyObject? } as NSArray
+        let gradient = CGGradientCreateWithColors(colorSpace, colors, nil)
+        // Apply gradient
+        CGContextClipToMask(context, rect, CGImage)
+        CGContextDrawLinearGradient(context,
+                                    gradient,
+                                    CGPoint(x: 0, y: 0),
+                                    CGPoint(x: 0, y: size.height),
+                                    CGGradientDrawingOptions(rawValue: 0))
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return image
+    }
+
+    /**
+     creates a radial gradient.
+
+     - parameter startColor: The start color
+     - parameter endColor: The end color
+     - parameter radialGradientCenter: The gradient center (default:0.5,0.5).
+     - parameter radius: Radius size (default: 0.5)
+     - parameter size: Image size (default: 100x100)
+     */
+    public convenience init?(startColor: UIColor,
+                             endColor: UIColor,
+                             radialGradientCenter: CGPoint = CGPoint(x: 0.5, y: 0.5),
+                             radius: CGFloat = 0.5,
+                             size: CGSize = CGSize(width: 100, height: 100)) {
```

Comment:
default size it's very strange. I think it can vary from screen to screen and from one project to another. nope?